sentinelhub/config.py

```diff
@@ -270,18 +254,9 @@ def get_config_dict(self) -> ConfigDict:
 
     @classmethod
     def get_config_location(cls) -> str:
-        """Returns location of configuration file on disk
-
-        :return: File path of `config.json` file
-        """
-        config_file = os.path.join(os.path.dirname(__file__), "config.json")
-
-        if not os.path.isfile(config_file):
-            with open(config_file, "w") as cfg_file:
-                default_dict = cls(use_defaults=True).get_config_dict()
-                json.dump(default_dict, cfg_file, indent=2)
-
-        return config_file
+        """Returns the default location of the user configuration file on disk."""
+        config_folder = platformdirs.user_config_dir("sentinelhub")
```

Comment:
I wanted to offload all decisions pertaining to that to `platformdirs` because otherwise we need to implement multiplatform tests for that, and I'd in general prefer not to. 

If you find any alternative libraries I'd be happy to switch. Or perhaps another solution on how to tell sh-py where to search for the config.

@batic any opinions as a Mac user?

Reply:
`dask` seems to be offloading the choice [here](https://github.com/dask/dask/blob/main/dask/config.py#L30) to 

```python
os.path.expanduser('~')
```

on Unix this is the same place, on Windows it's supposedly `C:/Users/<USERNAME>`

sounds sensible to me, if there was a custom folder in `~/.config/sentinelhub`.

Trying to check with some other common packages which support Windows as well

Reply:
> On Windows, HOME and USERPROFILE will be used if set, otherwise a combination of HOMEPATH and HOMEDRIVE will be used. An initial ~user is handled by stripping the last directory component from the created user path derived above.

Reply:
this way we can also drop the dependency

Reply:
Can confirm that this has the same behavior for me on Linux:
```python
    @classmethod
    def get_config_location(cls) -> str:
        """Returns the default location of the user configuration file on disk."""
        user_folder = os.path.expanduser("~")
        return os.path.join(user_folder, ".config", "sentinelhub", "config.json")
```

But is the `.config` subfolder suitable for Mac and (especially) Windows? Or do we need some other OS specific modifications?

Reply:
I have a feeling we might get some complaints by windows users over this, but we'll resolve it when we get there

Reply:
> But is the .config subfolder suitable for Mac and (especially) Windows?

for Mac it is, not sure for Windows, but if you're using Windows, you have other problems anyway, and in any case you can use the method here to get to the file. Additionally, the folder is hidden anyway, so they might not complain.